src/main/java/com/thevoxelbox/voxelsniper/wrapper/AsyncWorld.java

```diff
@@ -1250,6 +1226,71 @@ public Location locateNearestStructure(@NotNull Location arg0, @NotNull Structur
 		return parent.locateNearestStructure(arg0, arg1, arg2, arg3);
 	}
 
+	@Override
+	public @Nullable Location locateNearestBiome(@NotNull Location origin, @NotNull Biome biome, int radius) {
+		return parent.locateNearestBiome(origin, biome, radius);
+	}
+
+	@Override
+	public @Nullable Location locateNearestBiome(@NotNull Location origin, @NotNull Biome biome, int radius, int step) {
+		return parent.locateNearestBiome(origin, biome, radius, step);
+	}
+
+	@Override
+	public boolean isUltrawarm() {
+		return parent.isUltrawarm();
+	}
+
+	@Override
+	public boolean isNatural() {
+		return parent.isNatural();
+	}
+
+	@Override
+	public double getCoordinateScale() {
+		return parent.getCoordinateScale();
+	}
+
+	@Override
+	public boolean hasSkylight() {
+		return parent.hasSkylight();
+	}
+
+	@Override
+	public boolean hasBedrockCeiling() {
+		return parent.hasBedrockCeiling();
+	}
+
+	@Override
+	public boolean isPiglinSafe() {
+		return parent.isPiglinSafe();
+	}
+
+	@Override
+	public boolean doesBedWork() {
+		return parent.doesBedWork();
+	}
+
+	@Override
+	public boolean doesRespawnAnchorWork() {
+		return parent.doesRespawnAnchorWork();
+	}
+
+	@Override
+	public boolean hasRaids() {
+		return parent.hasRaids();
+	}
+
+	@Override
+	public boolean isFixedTime() {
+		return parent.isFixedTime();
+	}
+
+	@Override
+	public @NotNull Collection<Material> getInfiniburn() {
+		return parent.getInfiniburn();
+	}
+
```

Comment:
Unneeded and undesired. The Async* classes have been removed from Fawe since they are inherently unsafe and not recommendable to be used. I originally copied them here, because I didn't have the motivation to add it properly.
The proper way would be what Fawe does; use an EditSession instead of an AsyncWorld same should be done here and yeetus deletus all async classes.

Reply:
I'll take a look, some help with understanding EditSessions would be appreciated _(how to use them to follow your code styles and FAWE?)_.

Reply:
> Instead, what should I do, return null for every method, or an unsupported exception?

Yeetus deletus alongside the async classes. The override is needed, because these classes used to replace native methods delegating to our own implementation. Using an ES and deleting the classes will get a rid of the overrides

> I'll take a look, some help with understanding EditSessions would be appreciated (how to use them to follow your code styles and FAWE?).

Put them in a try with resources block, so they are closed properly when the the code exits the block, looks always a bit cleaner. But for a real "style" take a look at the Fawe code for ES's; our checkstyle plugin is not in use atm so no biggie but make sure it doesn't look out of place.

Reply:
Pheew, this creepy _migration_ is hopefully done, everything is now based on an editsession. I am committing my changes tomorrow, after a proofreading and tests.

Reply:
This request has been resolved and tried on a build server by several amateur players 😅 Hopefully nothing was broken.

**Note about undo code**: Since FAWE completely handles undo on its own side, may I _remove_/_comment out_ all undo related code?

Reply:
> Pheew, this creepy migration is hopefully done, everything is now based on an editsession. I am committing the changes tomorrow, after a proofreading and tests.

I'll give it a read at a later point.

> Note about undo code: Since FAWE completely handles undo on its own side, may I remove/comment out all undo related code?

Since we're now using an ES, undo/redo/history at all can be handled thru Fawe. Letting Fawe handle that makes named commands obsolete anyway.

> Also, with such changes, wouldn't a major version like 2.0.0, you suggested in another comment, be more adequate?

Likely. I created an addition branch, `feature/1.17.1` you can target this pr on allowing us to address several things and merging it all in one if ready.